File: src/main/java/edu/marmara/service/JsonService.java
package edu.marmara.service;

import com.fasterxml.jackson.core.JsonProcessingException;
import edu.marmara.config.Config;
import edu.marmara.model.Course;
import edu.marmara.model.Instructor;
import edu.marmara.model.Schedule;
import edu.marmara.model.Student;
import edu.marmara.model.Transcript;

import java.io.IOException;
import java.text.ParseException;
import java.util.List;

public interface JsonService {
    Student readStudentFromJson(String jsonFormattedStudent) throws ParseException, JsonProcessingException;

    List<Instructor> readInstructorsFromJson(String jsonFormattedInstructor) throws ParseException, JsonProcessingException;

    List<Course> readCoursesFromJson(String jsonFormattedCourse) throws ParseException, JsonProcessingException;

    Schedule readScheduleFromJson(String jsonFormattedSchedule) throws JsonProcessingException;

    Config readConfigFromJson(String jsonFormattedConfig) throws JsonProcessingException;

    void end() throws IOException;

    Transcript readTranscriptFromJson(String jsonFormattedTranscript) throws JsonProcessingException;
}

File: python/src/main/marmara/service/JsonService.py
from typing import List

from src.main.marmara.config.config import Config
from src.main.marmara.model.Course import Course
from src.main.marmara.model.Instructor import Instructor
from src.main.marmara.model.Schedule import Schedule
from src.main.marmara.model.Student import Student
from src.main.marmara.model.Transcript import Transcript


class JsonService:
    def read_student_from_json(self, json_formatted_student: str) -> Student:
        pass

    def read_instructors_from_json(self, json_formatted_instructor: str) -> List[Instructor]:
        pass

    def read_courses_from_json(self, json_formatted_course: str) -> List[Course]:
        pass

    def read_schedule_from_json(self, json_formatted_schedule: str) -> Schedule:
        pass

    def read_config_from_json(self, json_formatted_config: str) -> Config:
        pass

    def end(self):
        pass

    def read_transcript_from_json(self, json_formatted_transcript: str) -> Transcript:
        pass
